lib/Dialect/FIRRTL/FIRRTLAnnotationHelper.cpp

```diff
@@ -240,10 +240,16 @@ firrtl::resolveEntities(TokenAnnoTarget path, CircuitOp circuit,
         }
       if (!ref) {
         mlir::emitError(circuit.getLoc())
-            << "!cannot find port '" << field << "' in module "
+            << "cannot find port '" << field << "' in module "
             << target.moduleName();
         return {};
       }
+      if (isa<RefType>(ref.getType())) {
+        mlir::emitError(circuit.getLoc())
+            << "annotation cannot target reference port '" << field
+            << "' in module " << target.moduleName();
+        return {};
+      }
```

Comment:
A test of this would be good!

Reply:
I was silly and thought since the above path wasn't covered by test, this was utility code that was maybe not used due to the caching.  Should've checked.  Added tests for both, much better thanks!